consul/src/main/scala/io/buoyant/consul/v1/KvApi.scala

```diff
@@ -95,15 +107,30 @@ class KvApi(
     path: String,
     datacenter: Option[String] = None,
     cas: Option[String] = None,
+    recurse: Boolean = false,
     retry: Boolean = false
   ): Future[Boolean] = {
     val req = mkreq(
       http.Method.Delete,
       s"$kvPrefix$path",
       "cas" -> cas,
+      "recurse" -> (if (recurse) Some(recurse.toString) else None),
       "dc" -> datacenter
     )
-    executeJson[Boolean](req, retry).map(_.value)
+    execute[Boolean](req, retry).map(_.value)
   }
+}
+
+object Key {
+  def apply(key: String, value: String): Key = Key(Some(key), Some(Base64.getEncoder.encodeToString(value.getBytes)))
+}
 
+case class Key(
+  Key: Option[String],
```

Comment:
Those fields _must_ match the case of fields in JSON produced by Consul for mapper to be able to decode it. I'll add the comment explaining that. Other Consul APIs in the same package using the same approach - that's not 'my invention' but was done like that before my first PR. I'd be happy to change that to something else if you can advice me on another way of decoding JSON objects.